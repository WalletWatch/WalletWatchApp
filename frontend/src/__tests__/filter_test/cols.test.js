import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { shallow, configure } from 'enzyme';
import ColsFilter from '../../components/aggrid_filter/filters/cols_filter';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ColsFilter', () => {
    let mockApi;

    beforeEach(() => {
        mockApi = {
            getAllDisplayedColumns: jest.fn().mockReturnValue([
                { userProvidedColDef: { headerName: 'Column 1' }, colId: 'col1' },
                { userProvidedColDef: { headerName: 'Column 2' }, colId: 'col2' }
            ]),
            getColumnDefs: jest.fn().mockReturnValue([
                { field: 'col1', hide: false },
                { field: 'col2', hide: false }
            ]),
            setGridOption: jest.fn(),
            isDestroyed: jest.fn().mockReturnValue(false),
            addEventListener: jest.fn(),
            removeEventListener: jest.fn()
        };
    });

    it('renders correctly', () => {
        const { container } = render(
            <ColsFilter api={mockApi} />
        );

        expect(container).toMatchSnapshot();
    });

    it('handles checkbox click correctly', () => {
        let wrapper = shallow(<ColsFilter api={mockApi} />);
        
        expect(wrapper.find('#AllCols').prop('checked')).toBe(true);
        wrapper.find('#AllCols').simulate('change', { target: { checked: false } });
        expect(wrapper.find('#AllCols').prop('checked')).toBe(false);
    });
});



