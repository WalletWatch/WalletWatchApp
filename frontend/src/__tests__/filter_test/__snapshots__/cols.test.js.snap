// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColsFilter renders correctly 1`] = `
<div>
  <div
    class="column-filter"
    id="ag-cols"
  >
    <div
      style="overflow: hidden; height: 150px;"
    >
      <input
        class="title"
        id="searcherCols"
        placeholder="Search for cols..."
        type="text"
        value=""
      />
      <div
        class="column-filter-wrapper"
        id="searchCols"
        style="height: calc(100% - 28px);"
      >
        <div
          class="column-filter-item"
        >
          <input
            checked=""
            class="input-cols-filter"
            data-testid="AllCols"
            id="AllCols"
            type="checkbox"
            value="true"
          />
          <label
            class="input-column-name"
            for="AllCols"
          >
            (All)
          </label>
        </div>
        <div
          class="column-filter-item"
        >
          <input
            checked=""
            class="input-cols-filter"
            id="Column 1"
            type="checkbox"
            value="Column 1"
          />
          <label
            class="input-column-name"
            for="Column 1"
          >
            COLUMN 1
          </label>
        </div>
        <div
          class="column-filter-item"
        >
          <input
            checked=""
            class="input-cols-filter"
            id="Column 2"
            type="checkbox"
            value="Column 2"
          />
          <label
            class="input-column-name"
            for="Column 2"
          >
            COLUMN 2
          </label>
        </div>
      </div>
    </div>
    <div
      class="apply-button-wrapper"
    >
      <button
        class="apply-button"
      >
        Reset
      </button>
      <button
        class="apply-button"
        style="margin-left: 5px;"
      >
        Apply
      </button>
    </div>
  </div>
</div>
`;
